src/printer.js

```diff
@@ -1539,7 +1539,10 @@ function genericPrintNoParens(path, options, print) {
     case "IntersectionTypeAnnotation":
     case "UnionTypeAnnotation": {
       const types = path.map(print, "types");
-      const op = n.type === "IntersectionTypeAnnotation" ? "&" : "|";
+      const isIntersection = n.type === "IntersectionTypeAnnotation"
+      const op = isIntersection ? "&" : "|";
```

Comment:
you only use `op` where `isIntersection` is true, so you don't need this variable